design/FY2022/NFP-Python-HVAC-Diagram.md

```diff
@@ -0,0 +1,85 @@
+# Python HVAC Diagram #
+
+Jason DeGraw, Piljae Im, and Niraj Kunwar
+
+Oak Ridge National Laboratory
+
+Original Date: November 18, 2021
+
+## Justification for New Feature ##
+
+The HVAC-Diagram program is a Fortran utility that can be used to generate
+an SVG representation of the HVAC system in a model based on the BND file
+generated by EnergyPlus. The program has limited flexibility and the SVG
+output is not correctly rendered by all SVG rendering libraries. The SVG format
+[1] is a large and complex format with many capabilities but is somewhat
+overpowered for this application.
+
+## Overview ##
+
+A Python replacement for the utility will be developed to produce output similar
+to the functionality of the original utility. While SVG output will be considered,
+prioirity will be placed on production of output that conveys the same information
+in a more reliable manner.
+
+## Approach ##
+
+The BND output from EnergyPlus will be used to generate several forms of output:
+
+* One or more DOT language [2] files. This format is used by the
```

Comment:
I am not sure if the diagram appears in multiple output files that correspond to multiple DOT files that it will be as useful to the user. Users are looking for a single diagram that explains all the HVAC connections.

Reply:
Some of the programs that I have seen that can read DOT don't do great with multiple graphs in a single file, so adding some ability to control how much is in one file could be helpful, but not necessarily required.